File: frontend/app/src/main/java/com/example/commov/model/LoginModel.java
package com.example.commov.model;

public final class LoginModel {
    private final String email;
    private final String password;
    private final boolean passwordVisible;

    public LoginModel(String email, String password, boolean passwordVisible) {
        this.email = email;
        this.password = password;
        this.passwordVisible = passwordVisible;
    }

    public String getEmail() {
        return email;
    }

    public String getPassword() {
        return password;
    }

    public boolean isPasswordVisible() {
        return passwordVisible;
    }

    public LoginModel withEmail(String email) {
        return new LoginModel(email, password, passwordVisible);
    }

    public LoginModel withPassword(String password) {
        return new LoginModel(email, password, passwordVisible);
    }

    public LoginModel withPasswordVisible(boolean passwordVisible) {
        return new LoginModel(email, password, passwordVisible);
    }
}

File: frontend/app/src/main/java/com/example/commov/model/LoginModel.kt
package com.example.commov.model

data class LoginModel(
    val email: String,
    val password: String,
    val passwordVisible: Boolean
) {
    fun withEmail(email: String) = copy(email = email)
    fun withPassword(password: String) = copy(password = password)
    fun withPasswordVisible(passwordVisible: Boolean) = copy(passwordVisible = passwordVisible)
}
